app/models/spree/product.rb

```diff
@@ -288,6 +289,21 @@ def description=(html)
 
     private
 
+    def validate_variant_attrs
+      # Avoid running validation when we can't set variant attrs
+      # eg clone product. Will raise error if clonning a product with no variant
+      return if variants.first&.valid?
+
+      unless Spree::Taxon.find_by(id: primary_taxon_id)
+        errors.add(:primary_taxon_id,
+                   I18n.t('activerecord.errors.models.spree/product.must_exist'))
+      end
```

Comment:
I didn't follow why the category is included here, but I see you've been discussing with Gaetan so will let him confirm.